### PR TITLE
MOSIP-44155 : Junit PATCH /partners/{partnerId}/policies/{policyId}/a pi-keys/{apiKeyName} and updated stoplight documentation

### PR DESCRIPTION
**Pull request by Sudeep7353**
## Summary by CodeRabbit

* **Documentation**
  * Enhanced API documentation with comprehensive error code references and examples for OIDC client operations and API key management, including error scenarios and HTTP response examples.

* **Tests**
  * Added extensive test coverage for API key update functionality, including success scenarios with various field combinations and error handling for invalid inputs, permissions, and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: docs/partner_management_service_api_documentation.json [hunk 1]
```diff
@@ -4543,6 +4557,32 @@
...
+											"responseTime": "2024-11-05T10:36:13.780Z",
+											"metadata": {},
+											"response": null,
+											"errors": [
+												{
+													"errorCode": "PMS_ESI_010",
+													"message": "Invalid client authentication method provided."
+												}
+											]
 										}
```
[the remark this answers is not in the document]
Comment (Sudeep7353) [reply docs/partner_management_service_api_documentation.json]: @coderabbitai The response key will have null, when there is an error 

For example
{
  "id": null,
  "version": null,
  "metadata": null,
  "response": null,  // ← NULL HERE
  "responsetime": "2024-01-15T10:30:00.000Z",
  "errors": [
    {
      "errorCode": "PMS_PRT_001",
      "message": "Invalid request parameter - must not be null :partnerId"
    }
  ]
}

[the remark this answers is not in the document]
Comment (Sudeep7353) [reply docs/partner_management_service_api_documentation.json]: @coderabbitai check my comment above

[the remark this answers is not in the document]
Comment (Sudeep7353) [reply docs/partner_management_service_api_documentation.json]: Updated the PR @coderabbitai

**Review by mayuradesh — APPROVED** [review]
